test/unit/index.js

```diff
@@ -48,6 +41,55 @@ describe("asyncPool", function() {
         await asyncPool(2, [100, 500, 300, 200], timeout);
         expect(results).to.deep.equal([100, 300, 500, 200]);
       });
+
+      it("runs all promises in parallel when the pool is bigger than needed", async function() {
+        const results = [];
+        const timeout = i =>
+          new Promise(resolve =>
+            setTimeout(() => {
+              results.push(i);
+              resolve();
+            }, i)
+          );
+        await asyncPool(5, [100, 500, 300, 200], timeout);
+        expect(results).to.deep.equal([100, 200, 300, 500]);
+      });
+
+      it("rejects on error with pool limit effective", async function() {
+        const results = [];
+        const timeout = i =>
+          new Promise((resolve, reject) =>
+            setTimeout(() => {
+              results.push(i);
+
+              if (i === 300) {
+                reject(new Error("Oops"));
+              } else {
+                resolve();
+              }
+            }, i)
+          );
+        await rejects(asyncPool(2, [100, 500, 300, 200], timeout));
+      });
+
+      it("rejects on error when pool bigger than needed", async function() {
```

Comment:
Can we update this description to make the distinction between this and the above test more clear?

Reply:
Of course, we could do that, but I am not yet sure about the best way to go.
My original intention was to have two similar setups that only differ in pool size. In that case the descriptions should make clear that we are doing the same thing and that we expect the same output no matter what the pool size is.
After reading your comments I see the other possibility to have one spec that focuses on error handling (e.g. only throws for only one of the iterations), and another that is simpler and exists only to check for the absence of the UnhandledPromiseRejectionWarning. What would you prefer?

Reply:
I like your second approach, because (at least the way I understood it) they are simpler tests and each asserts for specific and different things